Jag uk/clarify initialization (#201)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

Multi-tenant support can be enabled through the use of identifiers in the `iss` claim, for example, `ts.example` may have a distinct Issuer identity for each sub domain, such as `customer1.ts.example` and `customer2.ts.example`.

### Registration Policies

Registration Policies refer to additional checks over and above the Mandatory Registration Checks that are performed before a Signed Statement is accepted to be registered to the Append-only Log.

Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.

Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log.

This specification leaves implementation, encoding and documentation of Registration Policies to the operator of the Transparency Service.

#### Mandatory Registration Checks

Transparency Services MUST, at a minimum, perform the following checks before registering a Signed Statement:
* Authenticate the Issuer of the Signed Statement

> To authenticate the Issuer of the Signed Statement the Transparency Service MUST refer to a trust anchor as defined in {{RFC6024}}: "A trust anchor represents an authoritative entity via a public key and associated data.

Reviewer: ```suggestion
The Transparency Service MUST authenticate the Issuer of Signed Statements by validating the COSE signature and checking the identity of the issuer through one of its configured trust anchors, using the `x5t` and `kid` headers in the protected header as hints. For instance, for X.509 signed claims the Transparency Service must validate a complete certificate chain from the certificate identified by `x5t` to one of the trusted root authority certificate of the Transparency Service.
```

Reviewer 2: yes

Reviewer 3: I will give it a 👍 but the use of the word is traipsing into configuration/policy territory that we have been trying to avoid.

The public key is used to verify digital signatures, and the associated data is used to constrain the types of information for which the trust anchor is authoritative."

> Typical representations of a trust anchor include certificates or raw public keys. The `x5t` and `kid` Claims in the protected header of Signed Statements can be used as hints for discovering trust anchors.

Reviewer 2: ```suggestion
Typical representations of a trust anchor include certificates or key material. The `x5t` and `kid` Claims in the protected header of Signed Statements can be used as hints for discovering trust anchors.
```

> Typical representations of a trust anchor include certificates or key material, e.g., a raw public key. The `x5t` and `kid` Claims in the protected header of Signed Statements can be used as hints for discovering trust anchors.

Reviewer: ```suggestion
````
now redundant with my proposed text

Before a Signed Statement is registered, the trust anchor used to verify its Issuer MUST be registered with the Transparency Service.

### Initialization and bootstrapping {#ts-initialization}

Since a Registration Policy is required prior to the registration of any Signed Statements, a means is required to configure the first Registration Policy that is not the standard issuance of a Signed Statement.
Transparency Services MUST support at least one of these methods:
* A built-in default Registration Policy
* Acceptance of a first Signed Statement whose payload is a valid Registration Policy, without performing registration checks

> * An out-of-band authenticated management interface

Reviewer 2: yes

## Transparency Service

> Transparency Services MUST feature an Append-only Log. The Append-only Log is the verifiable data structure which registers Signed Statements and supports the production of Receipts.

Reviewer 4: not to challenge our long standing tradition of using the term "Append-only log", but technically the Receipt depends on a verifiable data structure, of which an append only log (such as a merkle tree), is just one form.

> Transparency Services MUST feature an Append-only Log. The Append-only Log is the verifiable data structure which registers Signed Statements and supports the production of Receipts.

Reviewer 5: ```suggestion
Transparency Services MUST feature an Append-only Log.
The Append-only Log is the verifiable data structure that registers Signed Statements and supports the production of Receipts.
```

All Transparency Services MUST expose APIs for the registration of Signed Statements and issuance of Receipts.

Transparency Services MAY support additional APIs for auditing, for instance, to query the history of Signed Statements.

Typically a Transparency Service has a single Issuer identity which is present in the `iss` claim of Receipts for that service.

Multi-tenant support can be enabled through the use of identifiers in the `iss` claim, for example, `ts.example` may have a distinct Issuer identity for each sub domain, such as `customer1.ts.example` and `customer2.ts.example`.

### Registration Policies

Registration Policies refer to additional checks over and above the Mandatory Registration Checks that are performed before a Signed Statement is accepted to be registered to the Append-only Log.

Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.

> Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log and providing a means of fetching and verifying the corresponding Transparent Statement.

Reviewer 4: ```suggestion
Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log.
```

Reviewer 4: Simpler

Reviewer 5: :+1: Verification is covered elsewhere, minimizing the text needed here

Transparency Services MUST reject any attempt to register Signed Statements until at least one Registration Policy is registered.

> 

Reviewer 4: This is a deadlock... can't register a policy without a policy.

Reviewer 5: A registration policy is just another signed statement, so how do we release the deadlock?

Reviewer 5: I just noticed this [initialization proposal below](https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/201/files#diff-585fb424519b79cf000445f3425fa56d328cbaca5d2a622740ddc4e5ca91dbe1R412): 

> Since a Registration Policy is required prior to the registration of any Signed Statements, a means is required to configure the first Registration Policy that is not the standard issuance of a Signed Statement.
> Transparency Services MUST support at least one of these methods:

Author: I propose to resolve this comment given the presence of the Initialization section unless @OR13 still objects

> Registration Policies MAY be empty, in which case the Transparency Service only promises to enforce the Mandatory Registration Checks.

Reviewer 4: I think the distinction between Mandatory and Optional is more confusing here... I prefer to see all Policies as implementation specific.

Reviewer 5: Reviewing [draft 5](https://www.ietf.org/archive/id/draft-ietf-scitt-architecture-05.html), I couldn't find any text that states a receipt has a pointer to a registration policy, or another statement. 
Until we can clarify that, I'd suggest we remove this sentance
```suggestion
```

> This specification leaves implementation, encoding and documentation of Registration Policies to the operator of the Transparency Service.

Reviewer 4: This is all that needs to be said.

Reviewer 3: I support this recommendation.

Transparency Services MAY support additional APIs for auditing, for instance, to query the history of Signed Statements.

Typically a Transparency Service has a single Issuer identity which is present in the `iss` claim of Receipts for that service.

Multi-tenant support can be enabled through the use of identifiers in the `iss` claim, for example, `ts.example` may have a distinct Issuer identity for each sub domain, such as `customer1.ts.example` and `customer2.ts.example`.

### Registration Policies

Registration Policies refer to additional checks over and above the Mandatory Registration Checks that are performed before a Signed Statement is accepted to be registered to the Append-only Log.

Transparency Services MUST maintain Registration Policies which govern whether or not a given Signed Statement is eligible for registration.

Registration Policies MUST be made transparent and available to all clients of the Transparency Service by registering them as Signed Statements on the Append-only Log and providing a means of fetching and verifying the corresponding Transparent Statement.

> Transparency Services MUST reject any attempt to register Signed Statements until at least one Registration Policy is registered.

Reviewer 2: I am super curious. Can someone illustrate a scenario where it is bad to not have a registration policy (e.g., key material to check signatures) in the Append-only Log before allowing Signed Statements to be registered? This is an honest question.

> Transparency Services MUST reject any attempt to register Signed Statements until at least one Registration Policy is registered.

Reviewer 5: I don't think it's a question of being bad. It's a chick-egg scenario. If a registration policy is just a signed statement, then how do you get the first signed statement into other Append-only Log if all registrations must be evaluated against a registration policy (which is just another signed statement)?

> Transparency Services MUST reject any attempt to register Signed Statements until at least one Registration Policy is registered.

Reviewer 5: Based on this section, I believe Jon's resolved the deadlock: 
https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/201/files#diff-585fb424519b79cf000445f3425fa56d328cbaca5d2a622740ddc4e5ca91dbe1R409

Registration Policies MAY be empty, in which case the Transparency Service only promises to enforce the Mandatory Registration Checks.

This specification leaves implementation, encoding and documentation of Registration Policies to the operator of the Transparency Service.

#### Mandatory Registration Checks

> In addition to enforcing the rules contained in any registered Registration Policy, Transparency Services MUST, at a minimum, perform the following checks before registering a Signed Statement:

Reviewer 5: ```suggestion
Transparency Services MUST, at a minimum, perform the following checks before registering a Signed Statement:
```

Reviewer 5: minimizing

To authenticate the Issuer of the Signed Statement the Transparency Service MUST refer to a trust anchor as defined in {{RFC6024}}: "A trust anchor represents an authoritative entity via a public key and associated data.
The public key is used to verify digital signatures, and the associated data is used to constrain the types of information for which the trust anchor is authoritative."

Typical representations of a trust anchor include certificates or raw public keys. The `x5t` and `kid` Claims in the protected header of Signed Statements can be used as hints for discovering trust anchors.

> Before a Signed Statement is registered, the trust anchor used to verify its Issuer MUST be registered with the Transparency Service. To enable this, Transparency Services MUST support at least one of these methods:

Reviewer 5: ```suggestion
Before a Signed Statement is registered, the trust anchor used to verify its Issuer MUST be registered with the Transparency Service.
```

> * Inclusion of Issuer trust anchors in a registered Registration Policy

Reviewer 5: ```suggestion
```

> * Submission of Issuer trust anchors as a Signed Statement

Reviewer 5: ```suggestion
```

> * An out-of-band authenticated enrollment process

Reviewer 5: ```suggestion
```

> Before a Signed Statement is registered, the trust anchor used to verify its Issuer MUST be registered with the Transparency Service. To enable this, Transparency Services MUST support at least one of these methods:

Reviewer 5: Per [this comment](https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/201/files#r1488113904), I'd suggest removing lines 405-408

* An out-of-band authenticated enrollment process

### Initialization and bootstrapping {#ts-initialization}

> Since a Registration Policy is required prior to the registration of any Signed Statements, a means is required to configure the first Registration Policy that is not the standard issuance of a Signed Statement. Transparency Services MUST support at least one of these methods:

Reviewer 5: ```suggestion
Since a Registration Policy is required prior to the registration of any Signed Statements, a means is required to configure the first Registration Policy that is not the standard issuance of a Signed Statement.
Transparency Services MUST support at least one of these methods:
```
